cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -161,18 +168,16 @@ private void setResultSetMetadata(BatchSinkContext context) throws Exception {
    *
    * @throws Exception if the driver is not available
    */
-  private void ensureJDBCDriverIsAvailable(BatchSinkContext context) throws Exception {
+  private void ensureJDBCDriverIsAvailable() throws Exception {
     try {
       DriverManager.getDriver(dbSinkConfig.connectionString);
     } catch (SQLException e) {
-      // Load the plugin class to make sure it is available.
-      Class<?> driverClass = context.loadPluginClass(getJDBCPluginId());
-
       // Driver not found. We will try to register it with the DriverManager.
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
                 dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
                 JDBCDriverShim.class.getName());
-      DriverManager.registerDriver(new JDBCDriverShim((Driver) driverClass.newInstance()));
+      driverShim = new JDBCDriverShim(driverClass.newInstance());
+      DriverManager.registerDriver(driverShim);
```

Comment:
Also, maybe rename `deRegisterDriver` to `deRegisterAllDrivers`, or something better to reflect that all drivers with same classloader will get de-registered.?